algorithms/tree_isomorphisms.py

```diff
@@ -0,0 +1,209 @@
+from supporting_components.graph import Graph, Vertex
+from math import factorial
+
+
+def is_tree(G: "Graph"):
+    return len(G.edges) == len(G.vertices) - 1
+
+
+def trees_amount_of_isomorphisms(T1: "Graph", T2: "Graph"):
+    """
+    :param T1, T2: Graphs that are trees (so is_tree(T1) and is_tree(T2) return True)
+    :return:
+    """
+    root_T1 = root_label(T1)
+    root_T2 = root_label(T2)
+
+    # Tree can have one or two roots, if amount of roots is not equal, not isomorphic
+    if len(root_T1) != len(root_T2):
+        return False, 0
+
+    # Assign level numbers to all nodes
+    # todo: adept to when there are 2 roots.
+    assign_level(root_T1[0], 0, [])
+    assign_level(root_T2[0], 0, [])
+
+    # Assign all vertices to level number lists
+    L1 = {}
+    for v in T1:
+        L1.setdefault(v.level, []).append(v)
+    L2 = {}
+    for v in T2:
+        L2.setdefault(v.level, []).append(v)
+
+    # The number of levels of each tree should be equal
+    h1, h2 = max(L1.keys()), max(L2.keys())
+    if max(L1.keys()) != max(L2.keys()):
+        return False, 0
+    h = h1
+
+    for i in range(h, -1, -1):
+        H1 = []
```

Comment:
Ik kan niet helemaal volgens wat hier gebeurt, misschien is een beetje extra uitleg over wat de stappen doen niet verkeerd. Ook begrijp ik niet wat auto is van de vertex.

Reply:
Ik zal meer documentatie toevoegen